src/VisualStudio/Core/Def/Implementation/Progression/GraphQueries/SearchGraphQuery.cs

```diff
@@ -15,11 +16,29 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.Progression
 {
     internal sealed class SearchGraphQuery : IGraphQuery
     {
-        private readonly string _searchPattern;
+        private static readonly char[] DotCharacterArray = new[] { '.' };
+
+        private readonly string _lastPatternPart;
+        private readonly string _entireSearchPattern;
 
         public SearchGraphQuery(string searchPattern)
         {
-            _searchPattern = searchPattern;
+            _entireSearchPattern = searchPattern;
+
+            // If the pattern has a dot in it, then when we search the compilations, we actually
+            // want to only filter by the last name portion of hte pattern (as the compilers 
+            // only support returning the basic name of a symbol when searching and filtering 
+            // them).  Then, once we have all the results the compiler returned, we check and
+            // verify they match against the full dotted pattern.
+            if (searchPattern.Contains("."))
+            {
+                var patternParts = searchPattern.Split(DotCharacterArray, StringSplitOptions.RemoveEmptyEntries);
+                _lastPatternPart = patternParts.Last();
```

Comment:
Avoid the string.Split by using LastIndexOf.
Combine with the test on line 33.
And careful of leading, trailing and consecutive dots.